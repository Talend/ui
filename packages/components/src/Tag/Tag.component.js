import React from 'react';
import PropTypes from 'prop-types';
import { Tag as CoralTag } from '@talend/design-system';

/**
 * Proxy to https://design.talend.com/?path=/docs/components-tag--default-story#tag
 */
const Tag = ({ bsStyle, ...rest }) => {
	let StyledTag = CoralTag;
	switch (bsStyle?.toLowerCase()) {
		case 'info':
			StyledTag = CoralTag.Information;
			break;
		case 'success':
			StyledTag = CoralTag.Success;
			break;
		case 'warning':
			StyledTag = CoralTag.Warning;
			break;
		case 'danger':
			StyledTag = CoralTag.Destructive;
			break;
		default:
			break;
	}
	return <StyledTag {...rest} />;
};

Tag.displayName = 'Tag';

Tag.propTypes = {
	bsStyle: PropTypes.string,
};

export default Tag;
