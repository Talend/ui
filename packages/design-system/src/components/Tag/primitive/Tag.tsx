import React from 'react';

import style from './Tag.module.scss';

type TagProps = React.HTMLAttributes<HTMLSpanElement> & {
	label?: string;
};

const Tag = React.forwardRef(({ label, ...props }: TagProps, ref: React.Ref<HTMLSpanElement>) => {
	return (
		<span {...props} ref={ref} className={`${style.tag} ${props.className}`}>
			{label}
		</span>
	);
});

export default Tag;
