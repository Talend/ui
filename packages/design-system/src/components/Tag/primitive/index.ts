import TagPrimitive, { TagProps } from './TagPrimitive';

export type { TagProps };

export default TagPrimitive;
