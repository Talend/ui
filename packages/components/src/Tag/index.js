import Tag from './Tag.component';

export default Tag;
