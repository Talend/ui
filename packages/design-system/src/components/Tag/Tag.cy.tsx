/* eslint-disable testing-library/await-async-queries */
/* eslint-disable testing-library/prefer-screen-queries */
import { Tag } from '.';
import { TagVariant } from './Tag';

context('<Tag />', () => {
	it('should render', () => {
		cy.mount(<Tag data-testid="my.tag" />);
		cy.findByTestId('my.tag').should('to.exist');
	});
	it('should render variant information', () => {
		cy.mount(<Tag data-testid="my.tag" variant={TagVariant.information} />);
		cy.findByTestId('my.tag').should('to.exist');
	});
	it('should render variant destructive', () => {
		cy.mount(<Tag data-testid="my.tag" variant={TagVariant.destructive} />);
		cy.findByTestId('my.tag').should('to.exist');
	});
	it('should render variant success', () => {
		cy.mount(<Tag data-testid="my.tag" variant={TagVariant.success} />);
		cy.findByTestId('my.tag').should('to.exist');
	});
	it('should render variant warning', () => {
		cy.mount(<Tag data-testid="my.tag" variant={TagVariant.warning} />);
		cy.findByTestId('my.tag').should('to.exist');
	});
	it('should render variant beta', () => {
		cy.mount(<Tag data-testid="my.tag" variant={TagVariant.beta} />);
		cy.findByTestId('my.tag').should('to.exist');
	});
});
