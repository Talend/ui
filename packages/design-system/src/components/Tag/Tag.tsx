import React from 'react';

import * as S from './Tag.style';

export type TagProps = React.PropsWithChildren<any>;

// @see https://www.selbekk.io/blog/2020/05/forwarding-refs-in-typescript/
const Tag = React.forwardRef<HTMLSpanElement, TagProps>((props, ref) => {
	return <S.Tag ref={ref} {...props} />;
});

export default Tag;
