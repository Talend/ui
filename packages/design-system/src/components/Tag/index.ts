import Tag from './Tag';
import TagSuccess from './variations/Tag.success';
import TagDestructive from './variations/Tag.destructive';
import TagInformation from './variations/Tag.information';
import TagWarning from './variations/Tag.warning';

const TagComponent = Tag as typeof Tag & {
	Success: typeof TagSuccess;
	Destructive: typeof TagDestructive;
	Information: typeof TagInformation;
	Warning: typeof TagWarning;
};

TagComponent.Success = TagSuccess;
TagComponent.Destructive = TagDestructive;
TagComponent.Information = TagInformation;
TagComponent.Warning = TagWarning;

export default TagComponent;
