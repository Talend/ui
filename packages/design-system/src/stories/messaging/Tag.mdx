import { Controls, Meta, Canvas, Story } from '@storybook/blocks';
import { FigmaImage, Use } from '@talend/storybook-docs';

import { Tag } from '../../';
import { Status } from '../Status.block';

import * as Stories from './Tag.stories';

<Meta of={Stories} />
<Status id="tag" />

# Tag

The Tag component is useful to emphasize information to the user, works best with single word values.

## Zoning

<FigmaImage
	src="https://www.figma.com/file/wggW0Nf0ZujylGaXt5Snnn/Tags?node-id=1306%3A65"
	alt="zoning image for tags"
/>

## Style

<FigmaImage src="https://www.figma.com/file/wggW0Nf0ZujylGaXt5Snnn/Tags?node-id=0%3A43" alt="" />

### Variations

<Canvas>
	<Story of={Stories.Default} />
	<Story of={Stories.Information} />
	<Story of={Stories.Success} />
	<Story of={Stories.Warning} />
	<Story of={Stories.Destructive} />
	<Story of={Stories.Beta} />
</Canvas>

Or you can use variant props as well

<Canvas>
	<Story of={Stories.VariantDefault} />
	<Story of={Stories.VariantInformation} />
	<Story of={Stories.VariantSuccess} />
	<Story of={Stories.VariantWarning} />
	<Story of={Stories.VariantDestructive} />
	<Story of={Stories.VariantBeta} />
</Canvas>

## States

### Default

<Canvas>
	<Story of={Stories.Ellipsis} />
</Canvas>

## Content

Tags label UI items with keywords that help organize and categorize the items or draw more attention to them.

<Use>
	<Use.Do>
		<ul>
			<li>Use sentence case in tag labels.</li>
			<li>Use short labels on tags for easy scanning.</li>
			<li>Use clear and concise text.</li>
			<li>Use adjectives or nouns.</li>
		</ul>
	</Use.Do>
	<Use.Dont>
		<ul>
			<li>Don't use verbs.</li>
			<li>Don't use all caps in labels (except for acronyms).</li>
		</ul>
	</Use.Dont>
</Use>

## Interaction

N/A

## Accessibility

N/A

## Usage

<Canvas>
	<Story of={Stories.Usage} />
</Canvas>

<Controls of={Stories.Usage} />
