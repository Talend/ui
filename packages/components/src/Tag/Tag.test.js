import React from 'react';
import { shallow } from 'enzyme';
import Tag from './Tag.component';

describe('Tag', () => {
	const props = {
		children: 'The lazy quick brown fox jumps over the lazy dog',
	};

	it('should render', () => {
		const wrapper = shallow(<Tag {...props} />);
		expect(wrapper.getElement()).toMatchSnapshot();
	});
});
