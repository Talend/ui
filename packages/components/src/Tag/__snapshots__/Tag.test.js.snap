// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag should render 1`] = `
<Tag>
  The lazy quick brown fox jumps over the lazy dog
</Tag>
`;
